openc3/lib/openc3/interfaces/file_interface.rb

```diff
@@ -101,19 +109,30 @@ def read_interface
       while true
         if @file
           # Read more data from existing file
-          data = @file.read(@file_read_size)
+          if @file_read_size > 0
+            data = @file.read(@file_read_size)
+          else
+            data = @file.read
+          end
           if data and data.length > 0
             read_interface_base(data, nil)
-            return data, nil
+            return data, { filename: @filename, size: data.length }
```

Comment:
This allows the protocol to know when a file is opened and how much initial data was read